proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -380,18 +384,23 @@ private EnumValueDescriptor findValueByNameAndExtension(EnumDescriptor desc,
     }
   }
 
-  private static Method getCachedMethod(Class<?> clazz, String methodName)
-      throws NoSuchMethodException {
-    if (!mapOfMapOfMethods.containsKey(methodName)) {
-      mapOfMapOfMethods.put(methodName, new HashMap<Class<?>, Method>());
-    }
+  private static Method getCachedMethod(Class<?> clazz, String methodName,
+      Class<?>... methodParamTypes) throws NoSuchMethodException {
```

Comment:
Yeah, I think we're relying on the fact that the caller never uses overloads. It's cheating, but not bad cheating.